internal/client/authorization.go

```diff
@@ -7,74 +7,78 @@ import (
 	"sync/atomic"
 	"time"
 
+	"github.com/flightctl/flightctl/internal/auth/common"
 	"github.com/flightctl/flightctl/internal/cli/login"
 	"github.com/flightctl/flightctl/internal/util"
 	flightlog "github.com/flightctl/flightctl/pkg/log"
 	"github.com/sirupsen/logrus"
 )
 
 type accessTokenRefresher struct {
-	authInfo           AuthInfo
-	insecureSkipVerify bool
-	renewals           atomic.Int32
-	once               sync.Once
-	provider           login.AuthProvider
-	log                logrus.FieldLogger
+	config   *Config
+	renewals atomic.Int32
+	once     sync.Once
+	provider login.AuthProvider
+	log      logrus.FieldLogger
 }
 
 func (c *accessTokenRefresher) init() error {
-	switch c.authInfo.AuthType {
-	case "k8s":
-		c.provider = login.NewK8sOAuth2Config(c.authInfo.AuthCAFile, c.authInfo.ClientId, c.authInfo.AuthURL, c.insecureSkipVerify)
-	case "OIDC":
-		c.provider = login.NewOIDCConfig(c.authInfo.AuthCAFile, c.authInfo.ClientId, c.authInfo.AuthURL, c.insecureSkipVerify)
+	authInfo := c.config.AuthInfo
+	insecure := c.config.Service.InsecureSkipVerify
+	switch authInfo.AuthType {
```

Comment:
Maybe we should have a single place where the provider is created ?

Reply:
+1, I've moved the creation into separate function used by both `login.go` and `authorization.go`